### PR TITLE
ci: switch to taskgraph

### PR DESCRIPTION
**Pull request by jcristau**
This adds the following task kinds:
- docker-image: build images for other tasks (tox and skopeo) as well as
  images for the tooltool API/UI.
- tox: runs tests on various python version for the tooltool client and
  server
- push-image: injects version.json in the tooltool-api docker images,
  and pushes them to docker hub

### CODE REVIEW

**Comment by jcristau** [comment]
We'll need to update branch protection rules here due to the addition of `reporting: checks-v1`.

**Comment by jcristau** [comment]
Added `pr-complete` task for branch protection purposes.

File: taskcluster/docker/skopeo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+FROM debian:12-slim
+MAINTAINER Release Engineering <release@mozilla.com>
```
Comment (ahal) [inline taskcluster/docker/skopeo/Dockerfile:6]: nit: MAINTAINER is deprecated, this should be:
```
LABEL maintainer="Release ..."
```

File: taskcluster/kinds/docker-image/kind.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    client-tox-py38:
+        definition: tox
+        args:
+            BASE_IMAGE: python:3.8
+            SETUP_COMMAND: 'apt-get update && apt-get install -y zip xz-utils build-essential libssl-dev libffi-dev python3-dev'
+    client-tox-py39:
+        definition: tox
+        args:
+            BASE_IMAGE: python:3.9
+            SETUP_COMMAND: 'apt-get update && apt-get install -y zip xz-utils build-essential libssl-dev libffi-dev python3-dev'
```
Comment (ahal) [inline taskcluster/kinds/docker-image/kind.yml:22]: There's probably not much benefit in using tox here if we're going to split the tasks out one per Python version anyway. But that can be a follow-up if you want to reduce the simultaneous changes in this PR

**Review by ahal — APPROVED** [review]

**Review by gbrownmozilla — APPROVED** [review]

File: taskcluster/kinds/docker-image/kind.yml (hunk 2 above)
Comment (jcristau) [reply taskcluster/kinds/docker-image/kind.yml:22]: Right, I was aiming for the minimum amount of changes to reproduce the existing tasks (which is already quite a bit), I agree there's opportunities for later improvements.

File: taskcluster/docker/skopeo/Dockerfile (hunk 1 above)
Comment (jcristau) [reply taskcluster/docker/skopeo/Dockerfile:6]: Will fix, thanks.  (I copied this from taskgraph itself, but then made a couple of changes, so might as well get this in too.)

**Comment by jcristau** [comment]
I've updated the branch protection to require successful `pr-complete` task.

**Pull request merged**